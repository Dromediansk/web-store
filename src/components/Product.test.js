import { shallow } from "enzyme";
import React from "react";
import Product from "./Product";
import { shallowToJson } from "enzyme-to-json";

let output;

beforeEach(() => {
  const mockProps = {
    product: {
      id: 1,
      img: "imageSample",
      title: "titleOfProduct",
      price: 100,
      inCart: false
    }
  };
  output = shallow(<Product {...mockProps} />);
});

it("renders Product without crashing", () => {
  expect(output).toMatchSnapshot();
});
