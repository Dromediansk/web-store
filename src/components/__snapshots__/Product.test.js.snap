// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Product without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Product
    product={
      Object {
        "id": 1,
        "img": "imageSample",
        "inCart": false,
        "price": 100,
        "title": "titleOfProduct",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="card"
      >
        <Context.Consumer>
          [Function]
        </Context.Consumer>
        <div
          className="card-footer d-flex justify-content-between"
        >
          <p
            className="align-self-center mb-0"
          >
            titleOfProduct
          </p>
          <h5
            className="text-blue font-italic mb-0"
          >
            <span
              className="mr-1"
            >
              $
            </span>
            100
          </h5>
        </div>
      </div>,
      "className": "col-9 mx-auto col-md-6 col-lg-3 my-3",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Context.Consumer>
            [Function]
          </Context.Consumer>,
          <div
            className="card-footer d-flex justify-content-between"
          >
            <p
              className="align-self-center mb-0"
            >
              titleOfProduct
            </p>
            <h5
              className="text-blue font-italic mb-0"
            >
              <span
                className="mr-1"
              >
                $
              </span>
              100
            </h5>
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": [Function],
          },
          "ref": null,
          "rendered": [Function],
          "type": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": Object {
              "$$typeof": Symbol(react.context),
              "Consumer": [Circular],
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": null,
              "_currentRenderer2": null,
              "_currentValue": undefined,
              "_currentValue2": undefined,
              "_threadCount": 0,
            },
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="align-self-center mb-0"
              >
                titleOfProduct
              </p>,
              <h5
                className="text-blue font-italic mb-0"
              >
                <span
                  className="mr-1"
                >
                  $
                </span>
                100
              </h5>,
            ],
            "className": "card-footer d-flex justify-content-between",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "titleOfProduct",
                "className": "align-self-center mb-0",
              },
              "ref": null,
              "rendered": "titleOfProduct",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="mr-1"
                  >
                    $
                  </span>,
                  100,
                ],
                "className": "text-blue font-italic mb-0",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "$",
                    "className": "mr-1",
                  },
                  "ref": null,
                  "rendered": "$",
                  "type": "span",
                },
                100,
              ],
              "type": "h5",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
.card {
  border-color: transparent;
  transition: all 1s linear;
}

.card-footer {
  background: transparent;
  border-top: transparent;
  transition: all 1s linear;
}

&:hover {
  .card {
    border: 0.04rem solid rgba(0,0,0,0.2);
    box-shadow: 2px 2px 5px 0px rgba(0,0,0,0.2)
  }
  .card-footer {
    background: rgba(247, 247, 247);
  }
}

.img-container {
  position: relative;
  overflow: hidden
}

.card-img-top {
  transition: all 0.5s linear;
}

.img-container:hover .card-img-top {
  transform: scale(1.2);
}

.cart-btn {
  position: absolute;
  bottom: 0;
  right: 0;
  padding: 0.2rem 0.4rem;
  background: var(--lightBlue);
  border: none;
  color: var(--mainWhite);
  font-size: 1.4rem;
  border-radius: 0.5rem 0 0 0;
  transform: translate(100%, 100%);
  transition: all 0.5s linear;
}

.img-container:hover .cart-btn {
  transform: translate(0, 0);
}

.cart-btn:hover {
  color: var(--mainBlue);
  cursor: pointer;
}
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="card"
        >
          <Context.Consumer>
            [Function]
          </Context.Consumer>
          <div
            className="card-footer d-flex justify-content-between"
          >
            <p
              className="align-self-center mb-0"
            >
              titleOfProduct
            </p>
            <h5
              className="text-blue font-italic mb-0"
            >
              <span
                className="mr-1"
              >
                $
              </span>
              100
            </h5>
          </div>
        </div>,
        "className": "col-9 mx-auto col-md-6 col-lg-3 my-3",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Context.Consumer>
              [Function]
            </Context.Consumer>,
            <div
              className="card-footer d-flex justify-content-between"
            >
              <p
                className="align-self-center mb-0"
              >
                titleOfProduct
              </p>
              <h5
                className="text-blue font-italic mb-0"
              >
                <span
                  className="mr-1"
                >
                  $
                </span>
                100
              </h5>
            </div>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": [Function],
            },
            "ref": null,
            "rendered": [Function],
            "type": Object {
              "$$typeof": Symbol(react.context),
              "_calculateChangedBits": null,
              "_context": Object {
                "$$typeof": Symbol(react.context),
                "Consumer": [Circular],
                "Provider": Object {
                  "$$typeof": Symbol(react.provider),
                  "_context": [Circular],
                },
                "_calculateChangedBits": null,
                "_currentRenderer": null,
                "_currentRenderer2": null,
                "_currentValue": undefined,
                "_currentValue2": undefined,
                "_threadCount": 0,
              },
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="align-self-center mb-0"
                >
                  titleOfProduct
                </p>,
                <h5
                  className="text-blue font-italic mb-0"
                >
                  <span
                    className="mr-1"
                  >
                    $
                  </span>
                  100
                </h5>,
              ],
              "className": "card-footer d-flex justify-content-between",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "titleOfProduct",
                  "className": "align-self-center mb-0",
                },
                "ref": null,
                "rendered": "titleOfProduct",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="mr-1"
                    >
                      $
                    </span>,
                    100,
                  ],
                  "className": "text-blue font-italic mb-0",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "$",
                      "className": "mr-1",
                    },
                    "ref": null,
                    "rendered": "$",
                    "type": "span",
                  },
                  100,
                ],
                "type": "h5",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
.card {
  border-color: transparent;
  transition: all 1s linear;
}

.card-footer {
  background: transparent;
  border-top: transparent;
  transition: all 1s linear;
}

&:hover {
  .card {
    border: 0.04rem solid rgba(0,0,0,0.2);
    box-shadow: 2px 2px 5px 0px rgba(0,0,0,0.2)
  }
  .card-footer {
    background: rgba(247, 247, 247);
  }
}

.img-container {
  position: relative;
  overflow: hidden
}

.card-img-top {
  transition: all 0.5s linear;
}

.img-container:hover .card-img-top {
  transform: scale(1.2);
}

.cart-btn {
  position: absolute;
  bottom: 0;
  right: 0;
  padding: 0.2rem 0.4rem;
  background: var(--lightBlue);
  border: none;
  color: var(--mainWhite);
  font-size: 1.4rem;
  border-radius: 0.5rem 0 0 0;
  transform: translate(100%, 100%);
  transition: all 0.5s linear;
}

.img-container:hover .cart-btn {
  transform: translate(0, 0);
}

.cart-btn:hover {
  color: var(--mainBlue);
  cursor: pointer;
}
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
